src/libertem/udf/crystallinity.py

```diff
@@ -0,0 +1,84 @@
+import functools
+
+import numpy as np
+
+from libertem.common.buffers import BufferWrapper
+from libertem.masks import _make_circular_mask
+
+
+def make_result_buffers():
+    return {
+        'intensity': BufferWrapper(
+            kind="nav", dtype="float32"
+        ),
+    }
+
+
+def init_fft(partition, rad_in, rad_out, real_center, real_rad):
+    real_mask = 1-1*_make_circular_mask(
+        real_center[1], real_center[0], partition.shape.sig[1],
+        partition.shape.sig[0], real_rad
+    )
+    fourier_mask_out = 1*_make_circular_mask(
+        partition.shape.sig[1]*0.5, partition.shape.sig[0]*0.5,
+        partition.shape.sig[1], partition.shape.sig[0], rad_out
+    )
+    fourier_mask_in = 1*_make_circular_mask(
+        partition.shape.sig[1]*0.5, partition.shape.sig[0]*0.5,
+        partition.shape.sig[1], partition.shape.sig[0], rad_in
+    )
+    fourier_mask = fourier_mask_out - fourier_mask_in
+
+    kwargs = {
+        'real_mask': real_mask,
+        'fourier_mask': fourier_mask,
+    }
+    return kwargs
+
+
+def fft(frame, real_mask, fourier_mask, intensity):
+    intensity[:] = np.sum(np.fft.fftshift(abs(np.fft.fft2(frame*real_mask)))*fourier_mask)
```

Comment:
What do you think about using `pyfftw` here, if available?

As a note, it almost looks like we need to provide our own fft "abstraction" (very thin layer...), for example as `libertem.fft` and centralize the choice of fft backend. That is kind of stupid, but saves us from cluttering each FFT using module with the "optional import" pattern [as seen in the blobfinder UDF](https://github.com/LiberTEM/LiberTEM/blob/a3f54b1e531a248d74f0fe381b6df0068531c398/src/libertem/udf/blobfinder.py#L15-L28)...

Either that, or we use a single FFT impl. throughout. Anyways, the decision should not hold up merging this PR.

/cc @uellue